addons/web_editor/static/src/js/wysiwyg/widgets/link_tools.js

```diff
@@ -21,6 +20,10 @@ const LinkTools = Link.extend({
      * @override
      */
     init: function (parent, options, editable, data, $button, link) {
+        if (link && !$(link).is('a')) {
+            $(link).wrap('<a href="#"/>');
+            link = link.parentElement;
+        }
```

Comment:
I'm thrown off by the fact that `link` might be something else than a link... :p I suggest renaming the `link` function parameter to something like `node` then do:
```js
if (node && !$(node).is('a')) {
    $(node).wrap('<a href="#"/>');
    node = node.parentElement;
}
const link = node;
```